opus/application/static_media/js/browse.js

```diff
@@ -565,37 +565,23 @@ var o_browse = {
         let startObsLabel = o_browse.getStartObsLabel();
         let tab = opus.getViewTab();
         let contentsView = o_browse.getScrollContainerClass();
-        o_browse.currentOpusId = opusId;
+        o_browse.metadataDetailOpusId = opusId;
 
         let maxObs = (opus.prefs.view === "cart" ? o_cart.cartCount : opus.resultCount);
-        let obsNum = $(`${tab} .thumbnail-container[data-id=${opusId}]`).data("obs");
+        let element = (o_browse.isGalleryView() ? $(`${tab} .thumbnail-container[data-id=${opusId}]`) : $(`${tab} tr[data-id=${opusId}]`));
+        let obsNum = $(element).data("obs");
 
         let checkView = (direction === "next" ? obsNum <= maxObs : obsNum > 0);
 
         if (checkView) {
-            let element = (o_browse.isGalleryView() ? $(`${tab} .thumbnail-container[data-obs=${obsNum}]`) : $(`${tab} tr[data-obs=${obsNum}]`));
-            if (element.length === 0) {
-                // disable keydown on modal when it's loading
-                // this will make sure we have correct html elements displayed for prev observation
-                $("#galleryViewContents").addClass("op-disabled");
-                $(`${tab} ${contentsView}`).infiniteScroll("loadNextPage");
-
-                if (direction === "prev") {
-                    let startObs = obsNum - o_browse.getLimit();
-                    opus.prefs[startObsLabel] = (startObs > 0 ? startObs : 1);
-                } else {
-                    opus.prefs[startObsLabel] = obsNum;
-                }
-            } else {
-                // make sure the current element that the modal is displaying is viewable
-                if (!element.isOnScreen($(`${tab} .gallery-contents`))) {
-                    let startObs = $(`${tab} ${contentsView}`).data("infiniteScroll").options.obsNum;
-                    let rowCount = opus.getViewNamespace().galleryBoundingRect.x;
-                    let newStartObs = startObs + Math.floor((obsNum - startObs) / rowCount) * rowCount;
-                    let jsElement = $(`${tab} .thumbnail-container[data-id=${opusId}]`)[0];
-                    jsElement.scrollIntoView();
-                    o_browse.onUpdateSlider(newStartObs);
-                }
+            // make sure the current element that the modal is displaying is viewable
+            if (!element.isOnScreen($(`${tab} .gallery-contents`))) {
+                let startObs = $(`${tab} ${contentsView}`).data("infiniteScroll").options.obsNum;
+                let rowCount = opus.getViewNamespace().galleryBoundingRect.x;
+                let newStartObs = startObs + Math.floor((obsNum - startObs) / rowCount) * rowCount;
```

Comment:
This still doesn't work properly for binocs off the bottom of the screen. It jumps the scroll an entire page so that the binocs end up in the first row. This happens even if you just hit right arrow. I'm assuming we just want the gallery to scroll by a single row in this case. I really think you need to treat the obsNum < startObs and obsNum > startObs cases differently. You can't just to one formula that covers both.

Also, thanks to the joys of floating point, you can't rely on floor(x/y) to do what you want. You need flloat(x/y + 0.0000001) or some other very small number.

Reply:
Also there's something very weird with this in the table view. As you go backwards, you have to go one observation past the topmost one you can see before it triggers a scroll. Also, the amount of scroll when running off the top and when running off the bottom are totally different.